daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ComplexTypes.scala

```diff
@@ -54,6 +54,16 @@ sealed abstract class ComplexTypeBase(xmlArg: Node, parentArg: SchemaComponent)
     s
   }
 
+  modelGroup match {
```

Comment:
Everything else in this class is lazy or a def except for this. I'm not sure if there's a reason for that, but this being non-lazy will force evaluation of those things, which I'm not sure was intended. Maybe move this check into the `lazy val Seq(modelGroup)` block, something like `s(0) match { ...`? So it it will be lazily checked along with the other SDE in this class?